modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,758 @@
+        logger.info(`RtcMuteTimeout set to: ${this.rtcMuteTimeout}`);
+
+        /**
+         * This holds the timestamp indicating when remote video track
+         * entered forwarded sources set. Track entering forwardedSources
+         *  will have connection status restoring and when we start receiving
+         * video will become active, but if video is not received for certain
+         * time {@link DEFAULT_RESTORING_TIMEOUT} that track connection status
+         * will become interrupted.
+         *
+         * @type {number}
+         */
+        this.enteredForwardedSourcesTimestamp = null;
+
+        /**
+         * This holds the restoring timeout callback ID
+         * scheduled using window.setTimeout.
+         *
+         * @type {number}
+         */
+        this.restoringTimer = null;
+
+        /**
+         * This holds the current connection status (along with all the internal events that happen
+         * while in that state).
+         *
+         * The goal is to send this information to the analytics backend for post-mortem analysis.
+         */
+        this.connectionStatusMap = {};
+    }
+
+    /**
+     * Gets the video frozen timeout for given source name.
+     * @return {number} how long are we going to wait since RTC video muted
+     * even, before a video track is considered frozen.
+     * @private
+     */
+    _getVideoFrozenTimeout() {
+        const sourceName = this.track.getSourceName();
+
+        return this.rtc.isInForwardedSources(sourceName)
+            ? this.rtcMuteTimeout
+            : this.conference.isP2PActive() ? this.p2pRtcMuteTimeout : this.outOfLastNTimeout;
+    }
+
+    /**
+     * Initializes <tt>TrackStreamingStatus</tt> and bind required event
+     * listeners.
+     */
+    init() {
+        // Handles P2P status changes
+        this._onP2PStatus = this.figureOutConnectionStatus.bind(this);
+        this.conference.on(JitsiConferenceEvents.P2P_STATUS, this._onP2PStatus);
+
+        // Used to send analytics events for the participant that left the call.
+        this._onUserLeft = this.onUserLeft.bind(this);
+        this.conference.on(JitsiConferenceEvents.USER_LEFT, this._onUserLeft);
+
+        // On some browsers MediaStreamTrack trigger "onmute"/"onunmute"
+        // events for video type tracks when they stop receiving data which is
+        // often a sign that remote user is having connectivity issues
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+
+            this._onTrackRtcMuted = this.onTrackRtcMuted.bind(this);
+            this.rtc.addListener(
+                RTCEvents.REMOTE_TRACK_MUTE, this._onTrackRtcMuted);
+
+            this._onTrackRtcUnmuted = this.onTrackRtcUnmuted.bind(this);
+            this.rtc.addListener(
+                RTCEvents.REMOTE_TRACK_UNMUTE, this._onTrackRtcUnmuted);
+
+            // Listened which will be bound to JitsiRemoteTrack to listen for
+            // signalling mute/unmute events.
```

Comment:
In the legacy code, the `JitsiTrackEvents.TRACK_MUTE_CHANGED` and `JitsiTrackEvents.TRACK_VIDEOTYPE_CHANGED` event listeners are registered onto a track when the `JitsiConferenceEvents.TRACK_ADDED` event is emitted. This logic has been moved into the `init` method of `TrackStreamingStatus`.